pandas/core/arrays/masked.py

```diff
@@ -1414,3 +1414,32 @@ def _groupby_op(
         # res_values should already have the correct dtype, we just need to
         #  wrap in a MaskedArray
         return self._maybe_mask_result(res_values, result_mask)
+
+
+def transpose_homogenous_masked_arrays(masked_arrays):
+    """Transpose masked arrays in a list, but faster.
+
+    Input should be a list of 1-dim masked arrays of equal length and all have the
+    same dtype. The caller is responsible for ensuring validity of input data.
+    """
+    transposed_shape = len(masked_arrays), len(masked_arrays[0])
+    values = [arr._data for arr in masked_arrays]
+    transposed_values = np.empty(transposed_shape, dtype=values[0].dtype)
+    for i, val in enumerate(values):
+        transposed_values[i, :] = val
+    transposed_values = transposed_values.copy(order="F")
```

Comment:
this is faster than np.concatenate?

Reply:
It's about the same. I'll change to use np.concatenate, it's fewer lines.